plugins/modules/wafv2_rule_group.py

```diff
@@ -363,26 +371,33 @@ def main():
     if not metric_name:
         metric_name = name
 
-    rule_group = RuleGroup(module.client('wafv2'), name, scope, module.fail_json_aws)
+    wafv2 = module.client('wafv2')
+    rule_group = RuleGroup(wafv2, name, scope, module.fail_json_aws)
 
     change = False
     retval = {}
 
     if state == 'present':
         if rule_group.get():
-            change, rules = compare_priority_rules(rule_group.get().get('RuleGroup').get('Rules'), rules, purge_rules, state)
-            change = change or rule_group.get().get('RuleGroup').get('Description') != description
-
-            if change and not check_mode:
+            tagging_change = ensure_wafv2_tags(wafv2, rule_group.arn, tags, purge_tags,
+                                               module.fail_json_aws, module.check_mode)
+            rules_change, rules = compare_priority_rules(rule_group.get().get('Rules'), rules, purge_rules, state)
+            description_change = bool(description) and (rule_group.get().get('Description') != description)
+            change = tagging_change or rules_change or description_change
+            retval = rule_group.get()
+            if module.check_mode:
+                # In check mode nothing changes...
+                pass
```

Comment:
really small detail I know but we can check for `check_mode` prior to calling the previous line of `rule_group.get()`

Reply:
.get fetches the cached version

Reply:
ah! I was looking at get_group which calls refresh_group at the end. got it